src/channel.c

```diff
@@ -522,46 +526,57 @@ IMPLEMENT_MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_push, THANDLE(CHANNEL), ch
         /*Codes_SRS_CHANNEL_43_116: [ channel_push shall call srw_lock_acquire_exclusive. ]*/
         srw_lock_acquire_exclusive(channel_ptr->lock);
         {
-            /*Codes_SRS_CHANNEL_43_171: [ channel_push shall check if is_open is true. ]*/
-            if (!channel_ptr->is_open)
+            // BKTODO: state variable instead of flag
+            if (channel_ptr->is_overflowed != 0)
             {
-                /*Codes_SRS_CHANNEL_43_041: [ If there are any failures, channel_push shall fail and return CHANNEL_RESULT_ERROR. ]*/
-                LogError("channel is closing, cannot start a new channel_push operation");
-                result = CHANNEL_RESULT_ERROR;
+                LogInfo("channel_push: channel is overflowed, cannot push new data");
+                result = CHANNEL_RESULT_OVERFLOWED;
             }
             else
             {
-                /*Codes_SRS_CHANNEL_43_117: [ If the list of pending operations is empty or the first operation in the list of pending operations contains a non-NULL on_data_consumed_cb: ]*/
-                if (
-                    DList_IsListEmpty(&channel_ptr->op_list) ||
-                    CONTAINING_RECORD(channel_ptr->op_list.Flink, CHANNEL_OP, anchor)->on_data_consumed_cb != NULL
-                    )
+                if (interlocked_add(&channel_ptr->count_of_operations_pulled, 0) == 0 &&
+                    interlocked_add(&channel_ptr->overflow_level, 0) != 0 &&
```

Comment:
>interlocked_add(&channel_ptr->overflow_level, 0) [](http://example.com/codeflow?start=20&length=48)

I think it's better to use `UINT32_MAX` to mean no limit. 

Then this if condition can be simplified.